rs/ethereum/cketh/minter/src/eth_rpc.rs

```diff
@@ -718,10 +718,10 @@ pub enum ValidationError {
     UrlParseError(String),
     // #[error("hostname not allowed: {0}")]
     HostNotAllowed(String),
-    // #[error("credential path not allowed")]
-    CredentialPathNotAllowed,
-    // #[error("credential header not allowed")]
-    CredentialHeaderNotAllowed,
+    // #[error("API key not allowed")]
+    ApiKeyNotAllowed,
+    // #[error("header not allowed")]
+    HeaderNotAllowed,
```

Comment:
It's not clear to me that we even need those errors. Maybe something worth quickly discussing.

Reply:
It turns out that after reworking the validation logic, we can actually remove 4 of the validation error variants. Updated the PR to reflect this.

Reply:
nice!